4/Dockerfile

```diff
@@ -40,15 +40,53 @@ RUN set -eux; \
     apk del --purge .build-deps glibc-i18n; \
     rm -rf /tmp/*.apk /tmp/gcc /tmp/gcc-libs.tar.xz /tmp/libz /tmp/libz.tar.xz /var/cache/apk/*;
 
+#
+# AdoptOpenJDK/openjdk11 setup
+#
+RUN set -eux; \
```

Comment:
Why not just use an alpine based image from AdoptOpenJDK? like alpine-slim at https://hub.docker.com/r/adoptopenjdk/openjdk11/